### PR TITLE
docs(variants): Clearly explain variant notation

### PR DESCRIPTION
**Pull request by lachlanjc**
At @erikdstock’s excellent suggestion, clarifying how the notation for accessing variants works.

### CODE REVIEW

**Comment by erikdstock** [comment]
Looks great! I'm not sure how useful any of this would be useful for documentation but I'd like to clarify - styles are applied and overridden in this order:

1. `theme.styles.root`: automatically applied from the `body` element
2. `theme.styles.<tagName>` for MDX or `<Themed[tagName] / > or `Styled.[tagname]`  components
3. variants (probably best not to mix them):
  - built-in `variant` prop for theme-ui/components`sx` eg `<Button variant="outline"`
  - `sx` prop `variant` key with full path eg `<div sx={{variant: "buttons.outline"}}`
4. `sx` prop styles
5. [(maybe an anti-pattern)](https://github.com/system-ui/theme-ui/issues/839) `className` manually passed down from custom parent components, eg 
```tsx
const CustomButton = ({ children, className }) => (
  <Button
    className={className}
    variant="primary"
    sx={{ borderColor: "purple" }}
  >
    {children} (but custom)
  </Button>
)

// sx is converted to className
<CustomButton sx={{p: 5}}>Big Button</ CustomButton>
```

Finally, within the `theme-ui/components`: `variant` prop, which may have its own special key/default (eg `theme.buttons`), using a full theme path with the dot is an option as well- eg `<Button variant="secondary"` is equivalent to `variant="buttons.secondary"` and `<Button variant="cards.primary"` or `variant="styles.button"` would also be permitted, if not advisable.

[Example codesandbox](https://codesandbox.io/s/theme-ui-variants-and-sx-cc2ue?file=/pages/index.js)

I'm also wondering if there are any identified anti-patterns - In the example above radical departures like adding link styles to a button don't seem to work, and I've also found that there is always a balance to strike as far as what belongs isolated in a theme file vs declared where it is used or where some defaults are applied.

Another thought i'd be willing to take on - it would be nice to add doc comments for default theme keys to the custom components.

**Review by hasparus — APPROVED** [review]
Great improvement, folks.

I'm approving both the current addition and any changes / more content based on Erik's comment.

**Comment by lachlanjc** [comment]
Thanks @erikdstock! I added a section about `styles` linking to various places those are used. I’m going to merge this now as I don’t want making this perfect to interfere with getting something better live, but we could definitely keep making this better. Appreciate the feedback as always.

File: packages/docs/src/pages/guides/variants.mdx [hunk 1]
```diff
@@ -162,7 +165,19 @@ These variants can then be used in the `theme.styles` object, with the `sx` prop
 />
 ```
 
-## Themeable Layout Components
+## `styles` variants
+
+Variants inside the `styles` object, while usable through the same mechanisms as regular variants,
+are also used for other Theme UI APIs.
+
+- The entire [`styles` object](/theme-spec#styles) can style child HTML elements or MDX content,
```
Comment (hasparus) [inline packages/docs/src/pages/guides/variants.mdx:173]: Hey @lachlanjc, I'm not sure if this bullet doesn't "conflict" with the last one.
BaseStyles is used to style embedded HTML content, but MDX is styled with `Themed` (i.e. `##` maps to `Themed.h2`)

Comment (lachlanjc) [reply packages/docs/src/pages/guides/variants.mdx:173]: Hmm. Using Next.js that's not the case (MDX is treated the same as HTML children, so you need BaseStyles), so I've never experienced that. I guess we can just remove the MDX reference for now?

Comment (hasparus) [reply packages/docs/src/pages/guides/variants.mdx:173]: > MDX is treated the same as HTML children

That MDXProvider rendered by our ThemeProvider doesn't work in Next? (We pass an object with components through context)

Comment (lachlanjc) [reply packages/docs/src/pages/guides/variants.mdx:173]: Nope: https://github.com/system-ui/theme-ui/issues/1130

Comment (hasparus) [reply packages/docs/src/pages/guides/variants.mdx:173]: Doesn't a new version of MDX packages specified in resolutions solve that issue?

https://github.com/system-ui/theme-ui/issues/1130#issuecomment-678727533

Comment (lachlanjc) [reply packages/docs/src/pages/guides/variants.mdx:173]: Perhaps! Forgot about that & have never tried it

Comment (hasparus) [reply packages/docs/src/pages/guides/variants.mdx:173]: > I guess we can just remove the MDX reference for now?

Yeah, this sound like a good idea :D

Comment (lachlanjc) [reply packages/docs/src/pages/guides/variants.mdx:173]: https://github.com/system-ui/theme-ui/commit/fbb3451d43457ba041528c2bdb5ae9a602c8ff5b

**Comment by erikdstock** [comment]
Thanks yall, this is great!

**Pull request merged**